### PR TITLE
[trainer] Fix incorrect time values in JUnit reports

### LINKED ISSUE #29636: `trainer` incorrectly reports test duration for tests > 1 minute with Xcode 16
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
The junit reports generated by Fastlane when using Xcode 16 incorrectly report the test duration for tests longer than 1 minute. This is because xcresulttool outputs the duration in the format "1m 1s", while fastlane trainer assumes the duration is always in seconds.

##### Command executed
`run_tests`

##### Complete output when running fastlane, including the stack trace and command used

I will create a PR with tests that reproduce the issue.

### Environment

<details>
  <pre> | Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.228.0 | ✅ Up-To-Date  | </pre>
</details>

### PR DESCRIPTION
**Pull request by timbaev**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [ ] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

Resolves https://github.com/fastlane/fastlane/issues/29636

This change improves the accuracy and consistency of test duration reporting by switching from string-parsed durations to the numeric `durationInSeconds` field. It resolves an issue where the `duration` field could contain localized strings (e.g., `"1 мин 19 с"` in Russian), which previously caused incorrect duration parsing. The old parsing function only handled English strings and simple second values (e.g., `"45.5s"`), and did **not** support durations that included minutes, such as `"1m30s"` or `"1 мин 30 с"`. In such cases, it would return an incorrect value (e.g., `1.0` instead of the actual duration), thus impacting test reports. Now, by always using the `durationInSeconds` field, we ensure correct and locale-independent duration extraction.

---

### Description

- Refactored test case and repetition parsing logic to use the `durationInSeconds` field directly, replacing the previous approach of parsing the duration from a possibly localized string.
- Removed the unused and unreliable `parse_duration` method from test case attribute helpers.
- Updated existing JUnit fixture files (`Xcode16-Mixed-XCTest-SwiftTesting-WithRetries.junit` and `Xcode16-Mixed-XCTest-SwiftTesting-WithoutRetries.junit`) to reflect the more precise floating-point durations and suite times.
- Validated that these changes improve duration accuracy both in test results output and within fixture data.

### CODE REVIEW

**Comment by russellporter** [comment]
I was working on a fix at the same time 😅 . I figured out that the `durationInSeconds` field was only added in Xcode 16.3, so it might be good to keep the `duration` parsing approach as a fallback for the older versions.

Or maybe its possible to use the "legacy" reporting approach for older Xcode 16.x versions? I haven't looked into that possibility though.

**Comment by timbaev** [comment]
@russellporter Good catch! I’ve added a fallback for Xcode 16.0–16.2 with support for parsing durations with minutes.

File: trainer/lib/trainer/xcresult/test_case_attributes.rb [hunk 1]
```diff
@@ -19,6 +21,28 @@ def skipped?
       end
 
       module ClassMethods
+        def extract_duration(node)
+          if FastlaneCore::Helper.xcode_at_least?('16.3')
+            node['durationInSeconds']
```
Comment (russellporter) [inline trainer/lib/trainer/xcresult/test_case_attributes.rb]: I tested your patch out, this seems to return `nil` for some tests which then breaks fastlane further along as it assumes a non-nil duration. Maybe we can add a fallback to return 0.0 like was done for the old approach?

The problem seems to be that for tests that are recorded to take "0s", the "Test Case" entities don't have a duration. I guess its also a rounding error because the test did in fact run 🫠 

Example:
```json
{
  "children" : [
    {
      "details" : "iOS Simulator 18.5",
      "duration" : "0s",
      "durationInSeconds" : 0,
      "name" : "iPhone SE (3rd generation)",
      "nodeIdentifier" : "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
      "nodeType" : "Device",
      "result" : "Passed"
    },
    {
      "details" : "iOS Simulator 18.5",
      "duration" : "0s",
      "durationInSeconds" : 0,
      "name" : "iPhone SE (3rd generation)",
      "nodeIdentifier" : "YYYYYYYY-YYYY-YYYY-YYYY-YYYYYYYYYYYY",
      "nodeType" : "Device",
      "result" : "Passed"
    }
  ],
  "name" : "testMethodCallsInteractor()",
  "nodeIdentifier" : "TestServiceTests/testMethodCallsInteractor()",
  "nodeIdentifierURL" : "test://com.apple.xcode/bundle/shared-unit-tests/TestServiceTests/testMethodCallsInteractor()",
  "nodeType" : "Test Case",
  "result" : "Passed"
},
```

Comment (timbaev) [reply trainer/lib/trainer/xcresult/test_case_attributes.rb]: @russellporter Thanks for testing and catching this case! I agreed and pushed the fix 🤝

**Comment by timbaev** [comment]
@mollyIV Hello 👋 
Can you help with the review?

**Comment by iBotPeaches** [comment]
Sorry for delay. Looking here I have a few comments.

1. The build is failing so needs probably a merge/master or rebase to see if a real failure.
2. The description calls out modifications to junit sample files, but I don't see them.
3. As we introduce logic that acts differently based on Xcode version I believe we strongly need a spec test to confirm our assumptions since this logic with nil/fallback I'm slightly worried with.

**Comment by timbaev** [comment]
@iBotPeaches Thank you for the thorough review! I've addressed all the feedback:

**1. Build failure**
Already fixed with master merge/rebase.

**2. JUnit sample files**
Regenerated both fixture files with accurate floating-point durations:
- `Xcode16-Mixed-XCTest-SwiftTesting-WithRetries.junit`
- `Xcode16-Mixed-XCTest-SwiftTesting-WithoutRetries.junit`

The fixtures now reflect precise `durationInSeconds` values (e.g., `0.012162923812866211` instead of `0.012`), matching the actual data from Xcode 16.3+ xcresult bundles.

**3. Spec tests for version-based logic**
Added comprehensive spec tests in `trainer/spec/xcresult/test_case_attributes_spec.rb`:
- Tests for Xcode >= 16.3 using `durationInSeconds` field
- Tests for Xcode < 16.3 with enhanced string parsing (supports "1m 5s" format)
- Tests for nil/missing value handling with proper fallback to 0.0
- All edge cases covered (nil field, missing field, zero values, decimal values)

The spec tests verify both branches of the version check and ensure the nil fallback behavior works correctly as you noted.

**Comment by timbaev** [comment]
@iBotPeaches Hello 👋
Can you help with the review?
